Sources/Core/Client/Client+RxMessage.swift

```diff
@@ -0,0 +1,93 @@
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 06/02/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChatClient
+import RxSwift
+
+// MARK: Message Requests
+
+public extension Reactive where Base == Client {
+    
+    /// Get a message by id.
+    /// - Parameters:
+    ///   - messageId: a message id.
+    func message(with messageId: String) -> Observable<MessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
+            base.message(with: messageId, completion)
+        }))
+    }
+    
+    /// Mark all messages as read.
+    func markAllRead() -> Observable<EmptyData> {
+        connectedRequest(request({ [unowned base] completion in
+            base.markAllRead(completion)
+        }))
+    }
+    
+    /// Delete the message.
+    /// - Parameters:
+    ///   - message: a message for deleting.
+    func delete(message: Message) -> Observable<MessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
+            base.delete(message: message, completion)
+        }))
+    }
+    
+    /// Send a request for reply messages.
+    /// - Parameters:
+    ///   - message: a message.
+    ///   - pagination: a pagination (see `Pagination`).
+    func replies(for message: Message, pagination: Pagination) -> Observable<[Message]> {
+        connectedRequest(request({ [unowned base] completion in
+            base.replies(for: message, pagination: pagination, completion)
+        }))
+    }
+    
+    // MARK: - Reactions
+    
+    /// Add a reaction to the message.
+    /// - Parameters:
+    ///   - message: a message.
+    ///   - reactionType: a reaction type, e.g. like.
+    func addReaction(to message: Message, reactionType: ReactionType) -> Observable<MessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
+            base.addReaction(to: message, reactionType: reactionType, completion)
+        }))
+    }
+    
+    /// Delete a reaction to the message.
+    /// - Parameters:
+    ///   - message: a message.
+    ///   - reactionType: a reaction type, e.g. like.
+    func deleteReaction(from message: Message, reactionType: ReactionType) -> Observable<MessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
+            base.deleteReaction(from: message, reactionType: reactionType, completion)
+        }))
+    }
+    
+    // MARK: Flag Message
+    
+    /// Flag a message.
+    /// - Parameters:
+    ///   - message: a message.
+    func flag(message: Message) -> Observable<FlagMessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
+            base.flag(message: message, completion)
+        }))
+    }
+    
+    /// Unflag a message.
+    /// - Parameters:
+    ///   - message: a message.
+    func unflag(message: Message) -> Observable<FlagMessageResponse> {
+        connectedRequest(request({ [unowned base] completion in
```

Comment:
The structure
```swift
connectedRequest(request({ [unowned base] completion in
  base.unflag(message: message, completion)
}))
```
is repeated many times. I propose:
```swift
func connectedRequestWithUnownedBase<T>(_ block: ((Base, Completion) -> Void)) -> Observable<T> {
  connectedRequest(request({ [unowned base] completion in
    block(base, completion)
  }))
}

func unflag(message: Message -> Observable<FlagMessageResponse> {
  connectedRequestWithUnownedBase { base, completion in base.unflag(message: message, completion) }
}
```

The code is definitely not correct but you get the idea: hide the stuff behind a helper function to reduce repeating code.

Reply:
Did you check the original implementation?
```swift
func connectedRequest<T: Decodable>(_ rxRequest: Observable<T>) -> Observable<T> {
    base.isConnected ? rxRequest : connection.filter({ $0.isConnected }).take(1).flatMapLatest { _ in rxRequest }
}
```
It checks, if the Client is connected, do nothing, just return original request. And this will heppend in 99%
With your implementation we always need to create a block.

Reply:
We can rename `connectedRequest` to `connected` if you want.

Reply:
Yes, it works with `connected`!

Reply:
With my suggested implementation it just wraps the logic, to reduce repeated code. I'm still calling `connectedRequest`, please check again

Reply:
You are creating an unnecessary block `((Base, Completion) -> Void)` for each request.